Tcmp-SDK/Tcmp/CommandFamilies/System/IndicatorCommand.cs

```diff
@@ -0,0 +1,164 @@
+        /// </summary>
+        BlueLed,
+
+        /// <summary>
+        /// The buzzer.
+        /// </summary>
+        Buzzer
+    }
+
+    /// <summary>
+    /// Represents a command to modify the state of an indicator.
+    /// See <see cref="IndicatorOption"/> for available options.
+    /// </summary>
+    public abstract class IndicatorCommand : SystemCommand
+    {
+        /// <summary>
+        /// The command code for this IndicatorCommand.
+        /// </summary>
+        public abstract override byte CommandCode { get; }
+
+        /// <summary>
+        /// The target indicator.
+        /// </summary>
+        public abstract IndicatorOption Option { get; }
+    }
+
+    /// <summary>
+    /// Represents a command to activate (turn on) an indicator.
+    /// </summary>
+    public sealed class ActivateIndicator : IndicatorCommand
+    {
+        /// <inheritdoc cref="IndicatorCommand.CommandCode"/>
+        public override byte CommandCode { get; }
+
+        /// <inheritdoc cref="IndicatorCommand.Option"/>
+        public override IndicatorOption Option { get; }
+
+        /// <summary>
+        /// The amount of time (in milliseconds) to keep the indicator activated. 
+        /// </summary>
+        public UInt16 Duration { get; }
+
+        /// <summary>
+        /// Constructs an <see cref="ActivateIndicator"/> command that will activate the
+        /// specified <see cref="IndicatorOption"/> for an optional amount of time.
+        /// </summary>
+        /// <param name="option">The <see cref="IndicatorOption"/> to activate.</param>
+        /// <param name="duration">
+        /// The amount of time (in milliseconds) to keep the indicator
+        /// activated. A duration of `0` implies activation until a
+        /// <see cref="DeactivateIndicator"/> command is sent.
+        /// </param>
+        public ActivateIndicator(IndicatorOption option, UInt16 duration = 0)
+        {
+            Option = option;
+            Duration = (UInt16)duration;
+
+            switch (option)
+            {
+                case IndicatorOption.RedLed:
+                {
+                    CommandCode = 0x0A;
+                    break;
+                }
+                case IndicatorOption.GreenLed:
+                {
+                    CommandCode = 0x0D;
+                    break;
+                }
+                case IndicatorOption.BlueLed:
+                {
+                    CommandCode = 0x10;
+                    break;
+                }
+                case IndicatorOption.Buzzer:
+                {
+                    CommandCode = 0x13;
+                    break;
+                }
+                default:
+                    throw new NotImplementedException($"case {option} is not implemented");
+            }
+
+            if (Duration == 0) return;
+
+            // Duration is > 0, so modify command code and add as parameter
+            // TODO: Hard-coding this is not the most flexible solution
```

Comment:
No it's fine, this is clever and achieves a cleaner interface from the SDK user's perspective in that they can set duration to 0 indicating that they want it to be just activated indefinitely, or set a duration and use the "incremented command code".  The command codes will never change in the future, their set in the Tappy firmware so hard coding this is fine.